fili-core/src/test/groovy/com/yahoo/bard/webservice/web/JobsApiRequestSpec.groovy

```diff
@@ -232,6 +195,78 @@ class JobsApiRequestSpec extends Specification {
     def "getJobViews results in onError being called on it's observers if any of the JobRows in the ApiJobStore cannot be correctly mapped to a Job payload"() {
         setup:
         TestSubscriber<Map<String, String>> errorSubscriber = new TestSubscriber<>()
+
+        when:
+        defaultJobsApiRequest.getJobViews().subscribe(errorSubscriber)
+
+        then:
+        errorSubscriber.assertError(JobRequestFailedException.class)
+        errorSubscriber.getOnErrorEvents().get(0).getMessage() == "Jobs cannot be retrieved successfully due to internal error"
+    }
+
+    def "buildJobStoreFilter correctly parses a single filter query and returns ApiJobStoreFilter"() {
+        setup:
+        String filterQuery = "userId-eq[foo]"
+
+        when:
+        Set<ApiJobStoreFilter> filters = defaultJobsApiRequest.buildJobStoreFilter(filterQuery)
+
+        then:
+        filters.size() == 1
+        ApiJobStoreFilter filter = filters[0]
+        filter.jobField?.name == "userId"
+        filter.operation == ApiJobStoreFilterOperation.valueOf("eq")
+        filter.values == ["foo"] as Set
+    }
+
+    def "buildJobStoreFilter correctly parses multiple filter query"() {
+        setup:
+        String filterQuery = "userId-eq[foo],status-eq[successful]"
```

Comment:
We should probably use `success` rather than `successful` since `success` is a `DefaultJobStatus`, while `successful` is not, so that our test data is actually valid.